sdk/azidentity/azidentity.go

```diff
@@ -102,6 +129,27 @@ func newDefaultPipeline(o azcore.PipelineOptions) azcore.Pipeline {
 		azcore.NewRequestLogPolicy(o.LogOptions))
 }
 
+// NewDefaultMSIPipeline creates a Pipeline using the specified pipeline options needed
+// for a Managed Identity, such as a MSI specific retry policy
+func newDefaultMSIPipeline(o azcore.PipelineOptions) azcore.Pipeline {
+	if o.HTTPClient == nil {
+		o.HTTPClient = azcore.DefaultHTTPClientTransport()
+	}
+
+	// Set defaults needed for MSI retry policy
+	o.Retry.MaxTries = 5
+	o.Retry.Policy = azcore.RetryPolicyExponential
+	o.Retry.RetryDelay = 0
```

Comment:
Why retry delay of 0? Will the delay always be 0 in this case even with exponential backoff? i.e. 0^2 == 0

Reply:
I'm actually not using that variable. I'll delete it from the code.